railties/lib/rails/commands/warden.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails/commands/assets'
+require 'rails/commands/core'
+require 'rails/commands/dev_cache'
+require 'rails/commands/test'
+require 'rails/commands/tmp'
+
+module Rails
+  module Commands
+    # This class delegates each rails command to the corresponding class
+    class Warden
+      attr_accessor :command, :argv
+
+      def initialize(command, argv)
+        @command = command
+        @argv = argv
+      end
+
+      def run!
+        method_name = command.gsub(/:/, "_")
+
+        if Rails::Commands::Core::COMMAND_WHITELIST.include?(command)
```

Comment:
This whole if-statement is just silly. We need something more flexible than this.